src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.HashBucket.cs

```diff
@@ -30,7 +31,7 @@ internal enum OperationResult
         /// <summary>
         /// Contains all the keys in the collection that hash to the same value.
         /// </summary>
-        internal struct HashBucket
+        internal struct HashBucket : IEquatable<HashBucket>
```

Comment:
I guess I'll remove this, since throwing within the implementation doesn't make sense if I declare I support it.